src/ql/com/ddg/build.cc

```diff
@@ -427,7 +422,12 @@ class Builder {
         }
         if (!any_edge) {
             for (const auto &nc : global_write) {
-                QL_ASSERT(!nc.get().commutes_with(incoming));
+                if(nc.get().commutes_with(incoming)) {
```

Comment:
Is this there because sometimes the depgraph is circular?
Then that should be an issue. I still have to review depgraph construction, especially in the context of commuting gates. I suspect an error in the latter handling. Have you ever run a compiler with one of those commuting options on?

Reply:
No, during early testing the original QL_ASSERT triggered, but for other reasons than circularity. I replaced the assert by a somewhat more informational message. Jeroen solved the underlying problem, see https://github.com/QuTech-Delft/OpenQL/pull/427 and https://github.com/QuTech-Delft/OpenQL/pull/428 for background